app/src/main/java/openfoodfacts/github/scrachx/openfood/views/YourListedProducts.java

```diff
@@ -0,0 +1,329 @@
+                if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                    if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                        new MaterialDialog.Builder(this)
+                                .title(R.string.action_about)
+                                .content(R.string.permision_write_external_storage)
+                                .neutralText(R.string.txtOk)
+                                .show();
+                    } else {
+                        ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, Utils
+                                .MY_PERMISSIONS_REQUEST_STORAGE);
+                    }
+                } else {
+                    exportCSV();
+                }
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @OnClick(R.id.scanFirstYourListedProduct)
+    protected void onScanFirst() {
+        if (Utils.isHardwareCameraInstalled(getBaseContext())) {
+            if (ContextCompat.checkSelfPermission(getBaseContext(), Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+                if (ActivityCompat.shouldShowRequestPermissionRationale(YourListedProducts.this, Manifest.permission.CAMERA)) {
+                    new MaterialDialog.Builder(this)
+                            .title(R.string.action_about)
+                            .content(R.string.permission_camera)
+                            .neutralText(R.string.txtOk)
+                            .onNeutral((dialog, which) -> ActivityCompat.requestPermissions(YourListedProducts.this, new String[]{Manifest
+                                    .permission.CAMERA}, Utils.MY_PERMISSIONS_REQUEST_CAMERA))
+                            .show();
+                } else {
+                    ActivityCompat.requestPermissions(YourListedProducts.this, new String[]{Manifest.permission.CAMERA}, Utils
+                            .MY_PERMISSIONS_REQUEST_CAMERA);
+                }
+            } else {
+                Intent intent = new Intent(YourListedProducts.this, ContinuousScanActivity.class);
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                startActivity(intent);
+            }
+        }
+
+    }
+
+    public void setInfo(TextView view){
+        if(isEatenList){
+            view.setText(getString(R.string.txt_info_eaten_products));
+        } else{
+            view.setText(R.string.txt_info_your_listed_products);
+        }
+    }
+
+    public void exportCSV() {
+        boolean isDownload = false;
+        String folder_main = " ";
+        String appname = " ";
+        if ((BuildConfig.FLAVOR.equals("off"))) {
+            folder_main = " Open Food Facts ";
```

Comment:
Why the extra space before and after ?

Reply:
just followed the same pattern as in the exportCSV() of HistoryScanActivity as the functionality is  same :-)